From e19dda7ea692ec0c52c552ed9b1289656090df4b Mon Sep 17 00:00:00 2001
From: Yassine Oudjana <y.oudjana@protonmail.com>
Date: Mon, 23 Aug 2021 07:20:07 +0400
Subject: [PATCH] dt-bindings: power: supply: Add DT schema for qcom,fg
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add a DT schema for Qualcomm PMIC fuel gauge.

Signed-off-by: Yassine Oudjana <y.oudjana@protonmail.com>

Remove pmi8994 fuel gauge and update example for pmi8998-fg.

Signed-off-by: Barnabás Czémán <barnabas.czeman@mainlining.org>

Source-Commit: https://github.com/sdm660-mainline/linux/commit/e19dda7ea692ec0c52c552ed9b1289656090df4b
Scope: PMIC-common
Porting notes: Original binding applies unchanged to Linux 7.3-rc5.
---
diff --git a/Documentation/devicetree/bindings/power/supply/qcom,pmi8998-fg.yaml b/Documentation/devicetree/bindings/power/supply/qcom,pmi8998-fg.yaml
new file mode 100644
index 00000000000000..7514d6c0b5cfd9
--- /dev/null
+++ b/Documentation/devicetree/bindings/power/supply/qcom,pmi8998-fg.yaml
@@ -0,0 +1,71 @@
+# SPDX-License-Identifier: (GPL-2.0-only or BSD-3-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/power/supply/qcom,pmi8998-fg.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Qualcomm PMI8998 PMIC Fuel Gauge
+
+maintainers:
+  - Caleb Connolly <caleb@connolly.tech>
+  - Joel Selvaraj <jo@jsfamily.in>
+  - Yassine Oudjana <y.oudjana@protonmail.com>
+
+allOf:
+  - $ref: power-supply.yaml#
+
+properties:
+  compatible:
+    enum:
+      - qcom,pmi8998-fg
+
+  reg:
+    maxItems: 1
+
+  interrupts:
+    items:
+      - description: State of charge change interrupt
+
+  interrupt-names:
+    items:
+      - const: soc-delta
+
+  monitored-battery: true
+
+  power-supplies: true
+
+required:
+  - compatible
+  - reg
+  - monitored-battery
+  - interrupts
+  - interrupt-names
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/interrupt-controller/irq.h>
+
+    battery: battery {
+      compatible = "simple-battery";
+
+      charge-full-design-microamp-hours = <4070000>;
+      voltage-min-design-microvolt = <3400000>;
+      voltage-max-design-microvolt = <4400000>;
+    };
+
+    pmic {
+      #address-cells = <1>;
+      #size-cells = <0>;
+
+      fuel-gauge@4000 {
+        compatible = "qcom,pmi8998-fg";
+        reg = <0x4000>;
+
+        interrupts = <0x2 0x40 0x4 IRQ_TYPE_EDGE_RISING>;
+        interrupt-names = "soc-delta";
+
+        monitored-battery = <&battery>;
+      };
+    };

